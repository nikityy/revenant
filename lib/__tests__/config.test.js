const fs = require('fs');
const path = require('path');
const { promisify } = require('util');

const removeFile = promisify(fs.unlink);

const { readConfigFile, writeConfigFile } = require('../config');

const configPath = path.join(__dirname, 'config.json');
const tempConfigPath = path.join(__dirname, 'temp-config.json');
const unexistingConfigPath = path.join(__dirname, 'unexisting-config.json');

describe('readConfigFile', () => {
  test('should read config from a json file', async () => {
    const config = await readConfigFile(configPath);

    expect(config).toMatchSnapshot();
  });

  test('should fallback to empty config if file does not exist', async () => {
    const config = await readConfigFile(unexistingConfigPath);

    expect(config).toMatchSnapshot();
  });
});

describe('writeConfigFile', () => {
  afterEach(() => removeFile(tempConfigPath));

  test('should correctly serialize and write to file', async () => {
    const originalConfig = await readConfigFile(configPath);

    await writeConfigFile(tempConfigPath, originalConfig);

    const newConfig = await readConfigFile(tempConfigPath);

    expect(newConfig).toEqual(originalConfig);
  });
});
