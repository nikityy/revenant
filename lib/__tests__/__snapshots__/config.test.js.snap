// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readConfigFile should fallback to empty config if file does not exist 1`] = `
Object {
  "downloadPath": "~",
  "kinopoisk": Object {
    "listId": null,
    "usedId": null,
  },
  "rutracker": Object {
    "cookie": null,
  },
  "torrents": Object {},
  "version": "1",
  "watchlist": Array [],
}
`;

exports[`readConfigFile should read config from a json file 1`] = `
Object {
  "downloadPath": "~/Downloads/",
  "kinopoisk": Object {
    "listId": "3575",
    "usedId": "789114",
  },
  "rutracker": Object {
    "cookie": "bb-cookie:123",
  },
  "torrents": Object {
    "a": Array [
      "1",
      "2",
      "3",
    ],
    "b": Array [
      "4",
      "5",
    ],
    "c": Array [
      "6",
      "7",
      "1",
    ],
  },
  "version": "1",
  "watchlist": Array [
    "revenant",
    "who's afraid of virginia woolf",
  ],
}
`;
